test/scripts/e2e_subs/app-assets.sh

```diff
@@ -0,0 +1,273 @@
+${gcmd} clerk rawsend -f "$T/group.stx"
+
+[ "$(asset_ids "$SMALL")" = $ASSETID ] # asset ID
+[ "$(asset_bal "$SMALL")" = 999000 ]  # asset balance
+[ "$(asset_ids "$APPACCT")" = $ASSETID ]
+[ "$(asset_bal "$APPACCT")" = 1000 ]
+[ "$(balance "$SMALL")" =        994000 ] # 2 fees
+[ "$(balance "$APPACCT")" =      999000 ]
+
+# Withdraw 100 in app. Confirm that inner txn is visible to transaction API.
+TXID=$(appl "withdraw(uint64):void" --app-arg="int:100"  --foreign-asset="$ASSETID" --from="$SMALL" | app-txid)
+[ "$(rest "/v2/transactions/pending/$TXID" \
+        | jq '.["inner-txns"][0].txn.txn.aamt')" = 100 ]
+[ "$(rest "/v2/transactions/pending/$TXID?format=msgpack" | msgpacktool -d \
+        | jq '.["inner-txns"][0].txn.txn.type')" = '"axfer"' ]
+# Now confirm it's in blocks API (this time in our internal form)
+ROUND=$(rest "/v2/transactions/pending/$TXID" | jq '.["confirmed-round"]')
+rest "/v2/blocks/$ROUND" | jq .block.txns[0].dt.itx
+
+[ "$(asset_bal "$SMALL")" = 999100 ]   #  100 asset withdrawn
+[ "$(asset_bal "$APPACCT")" = 900 ] # 100 asset withdrawn
+[ "$(balance "$SMALL")" =        993000 ] # 1 fee
+[ "$(balance "$APPACCT")" =        998000 ] # fee paid by app
+
+appl "withdraw(uint64):void" --app-arg="int:100" --foreign-asset="$ASSETID"  --fee 2000 --from="$SMALL"
+[ "$(asset_bal "$SMALL")" = 999200 ]   #  100 asset withdrawn
+[ "$(balance "$SMALL")" = 991000 ]   # 2000 fee
+[ "$(asset_bal "$APPACCT")" = 800 ] # 100 asset  withdrawn
+[ "$(balance "$APPACCT")" = 998000 ] # fee credit used
+
+# Try to withdraw too much
+appl "withdraw(uint64):void" --app-arg="int:1000"  --foreign-asset="$ASSETID" --from="$SMALL"  && exit 1
+[ "$(asset_bal "$SMALL")" = 999200 ]   # no change
+[ "$(asset_bal "$APPACCT")" = 800 ]   # no change
+[ "$(balance "$SMALL")" = 991000 ]
+[ "$(balance "$APPACCT")" = 998000 ]
+
+# Show that it works AT exact asset balance
+appl "withdraw(uint64):void" --app-arg="int:800" --foreign-asset="$ASSETID" --from="$SMALL"
+[ "$(asset_bal "$SMALL")" = 1000000 ]
+[ "$(asset_bal "$APPACCT")" = 0 ]
+[ "$(balance "$SMALL")" = 990000 ]
+[ "$(balance "$APPACCT")" = 997000 ]
+
+USER=$(${gcmd} account new | awk '{ print $6 }') #new account
+${gcmd} clerk send -a 1000000 -f "$ACCOUNT" -t "$USER" #fund account
+asset-optin -f "$USER" -t "$USER"  --assetid "$ASSETID" #opt in to asset
+# SET $USER as clawback address
+${gcmd} asset config --manager $SMALL --assetid $ASSETID --new-clawback $USER
+cb_addr=$(${gcmd} asset info --assetid $ASSETID | clawback_addr)
+[ "$cb_addr" = "$USER" ]
+${gcmd} asset send -f "$SMALL" -t "$USER" -a "1000" --assetid "$ASSETID" --clawback "$USER"
+[ $(asset_bal "$USER") = 1000 ]
+[ $(asset_bal "$SMALL") = 999000 ]
+# rekey $USER to "$APPACCT"
+${gcmd} clerk send --from "$USER" --to "$USER" -a 0 --rekey-to "$APPACCT"
+# $USER should still have clawback auth. should have been authorized by "$APPACCT"
+${gcmd} asset send -f "$SMALL" -t "$USER" -a "1000" --assetid "$ASSETID" --clawback "$USER" && exit 1
```

Comment:
$USER has been rekeyed to $APPACCT.  Is there a way to sign this txn with $APPACCT or a different way to validate that $USER still has clawback auth?